src/rpc/util.cpp

```diff
@@ -179,11 +179,14 @@ std::string HelpExampleRpcNamed(const std::string& methodname, const RPCArgList&
 CPubKey HexToPubKey(const std::string& hex_in)
 {
     if (!IsHex(hex_in)) {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid public key: " + hex_in);
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Pubkey \"" + hex_in + "\" must be a hex string");
+    }
+    if (hex_in.length() != 66 && hex_in.length() != 130) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Pubkey \"" + hex_in + "\" must have a length of either 33 or 65 bytes");
```

Comment:
nit: not sure how often this HexToPubKey is called but I'm pretty sure
`hex_in.length() != 66 && hex_in.length() != 130`
will fail faster than
`!IsHex(hex_in)`
so it would be preferred to do that first

feel free to ignore this comment if that efficiency bump isn't a concern

Reply:
yup makes sense and I agree!